SSA/Core/WellTypedFramework.lean

```diff
@@ -207,19 +212,18 @@ def TSSA.eval {Op β : Type} [Goedel β] [TypedUserSemantics Op β] :
   {Γ : Context β} → {i : TSSAIndex β} → TSSA Op Γ i → (e : EnvC Γ) → i.eval
 | Γ, _, .assign lhs rhs rest, e =>
     rest.eval (fun
-      | _, Context.Var.prev v => e v
-      | _, Context.Var.last => rhs.eval e)
+      | _, { idx := ⟨i + 1, h⟩, of_type } => e { idx := ⟨i, Nat.lt_of_succ_lt_succ h⟩, of_type }
```

Comment:
I wonder if we can make this look cleaner using `@[match_pattern]` to create a pattern synonym for what we are matching here. The current match is quite a whirlwind of syntax.